src/test/java/seedu/address/logic/commands/DeleteAttendanceCommandTest.java

```diff
@@ -0,0 +1,169 @@
+import seedu.address.model.attendance.AttendanceRecord;
+import seedu.address.model.attendance.AttendanceRecordList;
+import seedu.address.model.attendance.Week;
+import seedu.address.model.lesson.Lesson;
+import seedu.address.model.moduleclass.ModuleClass;
+import seedu.address.model.student.Student;
+import seedu.address.testutil.LessonBuilder;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for
+ * {@code DeleteAttendanceCommand}
+ */
+public class DeleteAttendanceCommandTest {
+
+    private Model model = new ModelManager(getTypicalTutorsPet(), new UserPrefs());
+
+    @Test
+    public void constructor_nullIndexes_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new DeleteAttendanceCommand(
+                null, INDEX_FIRST_ITEM, INDEX_FIRST_ITEM, VALID_WEEK_1));
+        assertThrows(NullPointerException.class, () -> new DeleteAttendanceCommand(
+                INDEX_FIRST_ITEM, null, INDEX_FIRST_ITEM, VALID_WEEK_1));
+        assertThrows(NullPointerException.class, () -> new DeleteAttendanceCommand(
+                INDEX_FIRST_ITEM, INDEX_FIRST_ITEM, null, VALID_WEEK_1));
+    }
+
+    @Test
+    public void constructor_nullWeek_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new DeleteAttendanceCommand(
+                INDEX_FIRST_ITEM, INDEX_FIRST_ITEM, INDEX_FIRST_ITEM, null));
+    }
+
+    @Test
+    public void execute_validIndexes_success() {
+        Index moduleClassIndex = INDEX_FIRST_ITEM;
+        Index lessonIndex = INDEX_FIRST_ITEM;
+        Index studentIndex = INDEX_FIRST_ITEM;
+        Week targetWeek = VALID_WEEK_1;
+
+        ModuleClass moduleClass = model.getFilteredModuleClassList().get(moduleClassIndex.getZeroBased());
+        Student student = model.getFilteredStudentList().get(studentIndex.getZeroBased());
+        Lesson lesson = moduleClass.getLessons().get(lessonIndex.getZeroBased());
+        Map<UUID, Attendance> record = lesson.getAttendanceRecordList()
+                .getAttendanceRecord(targetWeek).getAttendanceRecord();
+        Map<UUID, Attendance> updatedRecord = new HashMap<>(record);
+        // delete week 1 attendance record
+        updatedRecord.remove(student.getUuid());
+        AttendanceRecord updatedAttendanceRecord = new AttendanceRecord(updatedRecord);
+        List<AttendanceRecord> updatedAttendanceRecords =
+                new ArrayList<>(lesson.getAttendanceRecordList().getAttendanceRecordList());
+        updatedAttendanceRecords.set(targetWeek.getZeroBasedWeekIndex(), updatedAttendanceRecord);
+        Lesson modifiedLesson = new LessonBuilder()
+                .withStartTime(lesson.getStartTime())
+                .withEndTime(lesson.getEndTime())
+                .withDay(lesson.getDay())
+                .withVenue(lesson.getVenue().venue)
+                .withAttendanceRecordList(new AttendanceRecordList(updatedAttendanceRecords))
+                .build();
```

Comment:
I think this can be simplified to 
```suggestion
        Lesson modifiedLesson = new LessonBuilder(lesson)
                .withAttendanceRecordList(new AttendanceRecordList(updatedAttendanceRecords)).build();
```

Reply:
Simplified for both ```AddAttendanceCommand``` and ```DeleteAttendanceCommand```